repos/system_upgrade/el8toel9/actors/sssdfacts/libraries/sssdfacts8to9.py

```diff
@@ -0,0 +1,53 @@
+from leapp.models import SSSDConfig8to9
+
+
+class SSSDFactsLibrary(object):
+    """
+    Helper library from SSSDFacts actor to allow unit testing.
+    """
+    def __init__(self, config):
+        self.config = config
+
+    def get_fallback(self, section, key, fallback):
+        if not self.config.has_option(section, key):
+            return fallback
+        return self.config.get(section, key)
+
+    def getboolean_fallback(self, section, key, fallback):
+        if not self.config.has_option(section, key):
+            return fallback
+        return self.config.getboolean(section, key)
+
+    def process(self):
+        """
+        Check SSSD configuration for the following options:
+
+        - sssd/enable_files_domain
+        - pam/pam_cert_auth
+        - explicit files provider domain present
+        """
+        facts = SSSDConfig8to9(
+            enable_files_domain_set=False,
+            explicit_files_domain=False,
+            pam_cert_auth=False,
+        )
+
+        facts.enable_files_domain_set = self.config.has_option('sssd', 'enable_files_domain')
+        facts.pam_cert_auth = self.getboolean_fallback('pam', 'pam_cert_auth', fallback=False)
```

Comment:
Can use configparser's native functions here

Reply:
@vinzenz You implemented custom fallback in order to support python2 as well. Is this still needed since the python2 tests are not run anymore?

Reply:
I did, yes and I actually started writing an answer to this already but forgot to send it how it looks like.

We shouldn't need it anymore, so we could revert those things, if you don't mind back to the python3 version.

Reply:
Done.